### PR TITLE
PGRedDefense

### PR DESCRIPTION
**Pull request by dMARLAN**
- "Red Mig-29" "Red F-14" etc. renamed to "Red One" "Red Two" etc. - Should make it easier for Mission Planners to edit, all they need to do is edit the aircraft inside the mission editor. They can edit loadouts, skills, aircraft type, etc. all from inside the editor.
- Each airbase will select a random aircraft out of the Red One, Two, etc. unless specified in the vars to use a specific aircraft.
- Code bloat removed. For pairs loop replaces most of the code.
- Code only works on PG for now, planning on updating it to work on all maps later.
- Some variables inside editor renamed, but updated inside the template to match.
- Template updated to work with script.
- GCI/CAP Stock increased to 4 & 8 respectively, allowing CAP/GCI to launch more than one wave of aircraft. Interval reduced drastically to compensate.
- GCI Strength randomized. Averages out to 0.4 strength currently (0.5 before) but ranges from 0.0 to 0.8 (range increases if random multiplier is increased) A strength of 0.4 should result in a GCI intercept of ~3 aircraft against a squadron of 8.

### CODE REVIEW

File: scripts/PGRedDefense.lua [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+local CAP_AIRCRAFT = {"Red One","Red Two","Red Three","Red Four","Red Five","Red Six"}
+local CAP_MAX_STOCK = 4
+local GCI_AIRCRAFT = {"Red One","Red Two","Red Three","Red Four","Red Five","Red Six"}
+local GCI_MAX_STOCK = 8
+local GCI_STRENGTH_MULT = 2 -- Incrementing multiplier increases strength by ~0.2 ; MULT of 2 is 0.4 strength.
+local border = border or true
+
+math.random()
+math.random()
+math.random()
```
Comment (kruzaavn) [inline scripts/PGRedDefense.lua:15]: what's the purpose of calling a random number 3 time but never asigining it to a value?

File: scripts/PGRedDefense.lua [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+local CAP_AIRCRAFT_LENGTH = 0
+for _ in pairs(CAP_AIRCRAFT) do CAP_AIRCRAFT_LENGTH = CAP_AIRCRAFT_LENGTH + 1 end
+local GCI_AIRCRAFT_LENGTH = 0
+for _ in pairs(GCI_AIRCRAFT) do GCI_AIRCRAFT_LENGTH = GCI_AIRCRAFT_LENGTH + 1 end
```
Comment (kruzaavn) [inline scripts/PGRedDefense.lua:20]: we should keep this DRY and write a function, or better see if one of the libraries we have access to has a function that determines the length of an array.

File: scripts/Super Carrier Recovery.lua [hunk 3]
```diff
@@ -150,8 +150,8 @@ repeat
   local RecoveryWindowEnd = sec2HHMM( MissionStart + PlannedTime + HHMM2sec(RecoveryWindowLength) )
 
   CarrierAirboss:AddRecoveryWindow(RecoveryWindowStart, RecoveryWindowEnd, 1, 0, TurnIntoWind, DeckWind)
-  MessageAll = MESSAGE:New(string.format("Recovery Window from %s to %s", RecoveryWindowStart, RecoveryWindowEnd), 120)
-  MessageAll:ToAll()
+  --MessageAll = MESSAGE:New(string.format("Recovery Window from %s to %s", RecoveryWindowStart, RecoveryWindowEnd), 120)
```
Comment (kruzaavn) [inline scripts/Super Carrier Recovery.lua:153]: any particular reason the message was commented out?

File: scripts/PGRedDefense.lua (hunk 1 above)
Comment (dMARLAN) [reply scripts/PGRedDefense.lua:15]: Assuming ED's seed is something smart like os.time() or the like (math.randomseed() is blocked), calling math.random() a few times makes the number more random. It's standard procedure for lua as far as I've seen.

File: scripts/Super Carrier Recovery.lua (hunk 3 above)
Comment (dMARLAN) [reply scripts/Super Carrier Recovery.lua:153]: It was just a giant block of text that was blocking things when I was testing, we can leave it in, forgot to uncomment it.

File: scripts/PGRedDefense.lua (hunk 2 above)
Comment (dMARLAN) [reply scripts/PGRedDefense.lua:20]: When I searched I didn't find any existing lua functions that do this, we could make a separate file with functions though.

Comment (kruzaavn) [reply scripts/PGRedDefense.lua:20]: I want to say that MOOSE has one

File: scripts/PGRedDefense.lua (hunk 1 above)
Comment (kruzaavn) [reply scripts/PGRedDefense.lua:15]: well if it's a constant seed to start you'll still wind up with the same number after 3 calls right?

Comment (kruzaavn) [reply scripts/PGRedDefense.lua:15]: if the seed is random or based on the system time then it's not necessary to call it multiple times.

File: scripts/PGRedDefense.lua (hunk 2 above)
Comment (dMARLAN) [reply scripts/PGRedDefense.lua:20]: I'll take a look while I'm at work

File: scripts/PGRedDefense.lua (hunk 1 above)
Comment (dMARLAN) [reply scripts/PGRedDefense.lua:15]: > But beware! The first random number you get is not really 'randomized' (at least in Windows 2K and OS X). To get better pseudo-random number just pop some random number before using them for real:
> 
> -- Initialize the pseudo random number generator
> math.randomseed( os.time() )
> math.random(); math.random(); math.random()
> -- done. :-)

File: scripts/PGRedDefense.lua (hunk 2 above)
Comment (dMARLAN) [reply scripts/PGRedDefense.lua]: found a lua function that works.

File: scripts/Super Carrier Recovery.lua (hunk 3 above)
Comment (dMARLAN) [reply scripts/Super Carrier Recovery.lua]: uncommented

**Pull request merged**